pkg/apis/batch/validation/validation.go

```diff
@@ -373,7 +373,7 @@ func ValidateJobUpdateStatus(job, oldJob *batch.Job) field.ErrorList {
 func ValidateJobSpecUpdate(spec, oldSpec batch.JobSpec, fldPath *field.Path, opts JobValidationOptions) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, ValidateJobSpec(&spec, fldPath, opts.PodValidationOptions)...)
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(spec.Completions, oldSpec.Completions, fldPath.Child("completions"))...)
+	allErrs = append(allErrs, validateIndexedJobCompletionsMutatedWithParallelism(spec, oldSpec, fldPath)...)
```

Comment:
Just looking this over, do you think we should add a test case that covers this?

Reply:
Yes, however, the [KEP](https://github.com/kubernetes/enhancements/pull/3724/files) has been updated with a different design (i.e. spec Completions can mutable iff spec.Parallelism is mutated in tandem with it such that spec.Completions == spec.Parallelism, thus allowing Indexed Jobs to be scaled by modifying completions & parallelism together). 

I am making a new PR which implements this behavior [here](https://github.com/kubernetes/kubernetes/pull/115236)